dashboard/lib/services/curriculum_pdfs.rb

```diff
@@ -121,15 +122,61 @@ def self.get_pdfless_lessons(script)
       end
     end
 
+    # Intended to be run to get all lesson plans from all stable units to the AI S3 bucket
+    def self.generate_lesson_plan_pdfs_for_all_stable_units
+      Unit.all.select(&:stable?).each do |unit|
+        generate_lesson_plan_pdfs_for_ai(unit)
+      end
+    end
+
+    def self.generate_lesson_plan_pdfs_for_ai(script)
```

Comment:
I tried to reuse this function in the `regenerate_pdfs` function below.  I couldn't quite finesse it enough and was trying to balance the need for `clean code` with the need to get this done :) 

I timeboxed the work and didn't get it quite right in that time - happy to take another look at it and spend more time to reduce repeated code.